lib/scholar/preprocessing/standard_scaler.ex

```diff
@@ -0,0 +1,145 @@
+defmodule Scholar.Preprocessing.StandardScaler do
+  @moduledoc """
+  Standardizes the tensor by removing the mean and scaling to unit variance.
+
+  #{~S'''
+  Formula for input tensor $x$:
+  $$
+  z = \frac{x - \mu}{\sigma}
+  $$
+  Where $\mu$ is the mean of the samples, and $\sigma$ is the standard deviation.
+  Standardization can be helpful in cases where the data follows
+  a Gaussian distribution (or Normal distribution) without outliers.
+  '''}
+
+  Centering and scaling happen independently on each feature by computing the relevant
+  statistics on the samples in the training set. Mean and standard deviation are then
+  stored to be used on new samples.
+  """
+
+  import Nx.Defn
+
+  @derive {Nx.Container, containers: [:standard_deviation, :mean]}
+  defstruct [:standard_deviation, :mean]
+
+  opts_schema = [
+    axes: [
+      type: {:custom, Scholar.Options, :axes, []},
+      doc: """
+      Axes to calculate the distance over. By default the distance
+      is calculated between the whole tensors.
+      """
+    ]
+  ]
+
+  @opts_schema NimbleOptions.new!(opts_schema)
+
+  @doc """
+  Compute the standard deviation and mean of samples to be used for later scaling.
+
+  ## Options
+
+  #{NimbleOptions.docs(@opts_schema)}
+
+  ## Return values
+
+    Returns a struct with the following parameters:
+
+    * `standard_deviation`: the calculated standard deviation of samples.
+
+    * `mean`: the calculated mean of samples.
+
+  ## Examples
+
+      iex> Scholar.Preprocessing.StandardScaler.fit(Nx.tensor([[1, -1, 2], [2, 0, 0], [0, 1, -1]]))
+      %Scholar.Preprocessing.StandardScaler{
+        standard_deviation: #Nx.Tensor<
+          f32[1][1]
+          [
+            [1.0657403469085693]
+          ]
+        >,
+        mean: #Nx.Tensor<
+          f32[1][1]
+          [
+            [0.4444444477558136]
+          ]
+        >
+      }
+  """
+  deftransform fit(tensor, opts \\ []) do
+    NimbleOptions.validate!(opts, @opts_schema)
+    {std, mean} = fit_n(tensor, opts)
+
+    %__MODULE__{standard_deviation: std, mean: mean}
+  end
+
+  defnp fit_n(tensor, opts) do
+    std = Nx.standard_deviation(tensor, axes: opts[:axes], keep_axes: true)
+    mean_reduced = Nx.mean(tensor, axes: opts[:axes], keep_axes: true)
+    mean_reduced = Nx.select(std == 0, 0.0, mean_reduced)
+
+    {std, mean_reduced}
+  end
+
+  @doc """
+  Performs the standardization of the tensor using a fitted scaler.
+
+  ## Examples
+
+      iex> t = Nx.tensor([[1, -1, 2], [2, 0, 0], [0, 1, -1]])
+      iex> scaler = Scholar.Preprocessing.StandardScaler.fit(t)
+      %Scholar.Preprocessing.StandardScaler{
+        standard_deviation: #Nx.Tensor<
+          f32[1][1]
+          [
+            [1.0657403469085693]
+          ]
+        >,
+        mean: #Nx.Tensor<
+          f32[1][1]
+          [
+            [0.4444444477558136]
+          ]
+        >
+      }
```

Comment:
```suggestion
      iex> t = Nx.tensor([[1, -1, 2], [2, 0, 0], [0, 1, -1]])
      iex> scaler = Scholar.Preprocessing.StandardScaler.fit(t)
      %Scholar.Preprocessing.StandardScaler{
        standard_deviation:  Nx.tensor(
          [
            [1.0657403469085693]
          ]
        ),
        mean:  Nx.tensor(
          [
            [0.4444444477558136]
          ]
        )
      }
```

Reply:
Generally, if you want to execute a test with a structure as a result, then you need to pass tenors as you would create them in `defn`. So you should replace all `#Nx.Tensor<...>` with `Nx.tensor(..)`